packages/mds-ingest-service/repository/index.ts

```diff
@@ -473,21 +473,23 @@ class IngestReadWriteRepository extends ReadWriteRepository {
   public getLatestTelemetryForDevices = async (device_ids: UUID[]): Promise<TelemetryDomainModel[]> => {
     try {
       const connection = await this.connect('ro')
+      /* sets enable_bitmapscan = off for this connection the duration of the session */
+      await connection.getRepository(TelemetryEntity).query(`select set_config('enable_bitmapscan','off', false)`)
+
       const entities = await connection
         .createQueryBuilder(TelemetryEntity, 'telemetry')
-        .innerJoin(
-          subquery =>
-            subquery
-              .select('id, RANK() OVER (PARTITION BY device_id ORDER BY timestamp DESC) AS rownum')
-              .where('device_id = ANY(:device_ids)', { device_ids })
-              .from(TelemetryEntity, 't'),
-          'last_device_telemetry',
-          'last_device_telemetry.id = telemetry.id AND last_device_telemetry.rownum = 1'
-        )
+        .distinctOn(['device_id'])
+        .orderBy('device_id', 'DESC')
+        .addOrderBy('timestamp', 'DESC')
         .getMany()
       return entities.map(TelemetryEntityToDomain.mapper())
     } catch (error) {
       throw RepositoryError(error)
+    } finally {
+      const connection = await this.connect('ro')
+      /* sets enable_bitmapscan = on for this connection the duration of the session. Seems like it shouldn't be an issue,
+         but connection pooling could/should be in use, and persistent connections are used between queries. */
+      await connection.getRepository(TelemetryEntity).query(`select set_config('enable_bitmapscan','on', false)`)
```

Comment:
```suggestion
      await connection.getRepository(TelemetryEntity).query(`select set_config('enable_bitmapscan','on', true)`)
```

Reply:
yeah, "true" means transaction only, rather than session only, which now works w/ a good transaction-api for typeORM